File: src/test/java/plus/maa/backend/repository/GithubRepositoryTest.java
package plus.maa.backend.repository;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import plus.maa.backend.config.external.MaaCopilotProperties;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertNotNull;

@SpringBootTest
class GithubRepositoryTest {

    @Autowired
    private GithubRepository repository;
    @Autowired
    private MaaCopilotProperties properties;

    @Test
    public void testGetTrees() {
        var maaTree = repository.getTrees(properties.getGithub().getToken(), "d989739981db071e80df1c66e473c729b50e8073");
        assertNotNull(maaTree);
    }

    @Test
    public void testGetCommits() {
        var commits = repository.getCommits(properties.getGithub().getToken());
        assertNotNull(commits);
        assertFalse(commits.isEmpty());
    }

    @Test
    void testGetContents() {
        var contents = repository.getContents(properties.getGithub().getToken(),"");
        assertNotNull(contents);
        assertFalse(contents.isEmpty());
    }
}

File: src/test/kotlin/plus/maa/backend/repository/GithubRepositoryTest.kt
package plus.maa.backend.repository

import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import plus.maa.backend.config.external.MaaCopilotProperties

@SpringBootTest
class GithubRepositoryTest(
    @Autowired val repository: GithubRepository,
    @Autowired val properties: MaaCopilotProperties
) {
    @Test
    fun testGetTrees() {
        repository.getTrees(properties.github.token, "d989739981db071e80df1c66e473c729b50e8073")
    }

    @Test
    fun testGetCommits() {
        val commits = repository.getCommits(properties.github.token)
        check(commits.isNotEmpty())
    }

    @Test
    fun testGetContents() {
        val contents = repository.getContents(properties.github.token, "")
        check(contents.isNotEmpty())
    }
}